semgrep/semgrep/core_runner.py

```diff
@@ -95,6 +118,25 @@ def _run_rules(
         outputs: List[PatternMatch] = []  # multiple invocations per language
         errors: List[Any] = []
 
+        # This will need to be addressed in the future. Since we flatten all the patterns,
+        # there's no way to tell semgrep which equivalences apply to which rules.
+        # So, my approach here is a naive implementation which likewise flattens all equivalences...
+        # Here there be dragons.... :-(
+        #  .>   )\;`a__
+        # (  _ _)/ /-." ~~
+        #  `( )_ )/
+        #   <_  <_ sb/dwb
+        equivalences = self._flatten_all_equivalences(rules)
+        if equivalences:
+            equiv_fout = tempfile.NamedTemporaryFile("w")
```

Comment:
Can we use `with` context similar to line 143 below? So we dont have to remember to close.

Reply:
Sure; I went with this method because I thought it would be easier to read than two separate with contexts. I'll fix it up.